### ISSUE TITLE
haskellPackages.vector broken

### ISSUE THREAD

**Issue by NeQuissimus** [NixOS/nixpkgs#117792]
**Describe the bug**
Package cannot be built

**To Reproduce**
1. `nix-build -A haskellPackages.vector`

**Expected behavior**
Package builds properly

**Additional context**
I found this as a transitive dependency for `exa`

Build log:

```
shrinking /nix/store/nmhd3zwhnp7avk1bx0vg157sssgybzyj-connection-0.3.1/lib/ghc-8.10.4/x86_64-linux-ghc-8.10.4/libHSconnection-0.3.1-5rmTUPwgpun9eoemLax2RI-ghc8.10.4.so
strip is /nix/store/xdii8qvch5h8chyp0z2is2qzky565w68-binutils-2.35.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/nmhd3zwhnp7avk1bx0vg157sssgybzyj-connection-0.3.1/lib
patching script interpreter paths in /nix/store/nmhd3zwhnp7avk1bx0vg157sssgybzyj-connection-0.3.1
checking for references to /build/ in /nix/store/nmhd3zwhnp7avk1bx0vg157sssgybzyj-connection-0.3.1...
shrinking RPATHs of ELF executables and libraries in /nix/store/k1gp268g31carcc290rx4ihkg2ilww4l-connection-0.3.1-doc
strip is /nix/store/xdii8qvch5h8chyp0z2is2qzky565w68-binutils-2.35.1/bin/strip
patching script interpreter paths in /nix/store/k1gp268g31carcc290rx4ihkg2ilww4l-connection-0.3.1-doc
checking for references to /build/ in /nix/store/k1gp268g31carcc290rx4ihkg2ilww4l-connection-0.3.1-doc...
/nix/store/6c47azxacncswc1pllzj28zfzqw40d7c-stdenv-linux/setup: line 1309:   221 Killed                  ./Setup build
error: builder for '/nix/store/ki28i4frhflavmvxb48bf9d4a36d1ymd-vector-0.12.2.0.drv' failed with exit code 137;
       last 10 log lines:
       >
       > tests/Tests/Vector/Property.hs:52:1: warning: [-Wunused-imports]
       >     The import of ‘Data.Functor.Identity’ is redundant
       >       except perhaps to import instances from ‘Data.Functor.Identity’
       >     To import instances alone, use: import Data.Functor.Identity()
       >    |
       > 52 | import Data.Functor.Identity
       >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       > [ 5 of 12] Compiling Tests.Vector.Unboxed ( tests/Tests/Vector/Unboxed.hs, dist/build/vector-tests-O2/vector-tests-O2-tmp/Tests/Vector/Unboxed.o, dist/build/vector-tests-O2/vector-tests-O2-tmp/Tests/Vector/Unboxed.dyn_o )
       > /nix/store/6c47azxacncswc1pllzj28zfzqw40d7c-stdenv-linux/setup: line 1309:   221 Killed                  ./Setup build
       For full logs, run 'nix log /nix/store/ki28i4frhflavmvxb48bf9d4a36d1ymd-vector-0.12.2.0.drv'.
error: 1 dependencies of derivation '/nix/store/nk3rz1yvbsig58795cj5ma2dayi7cjqb-JuicyPixels-3.3.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/14d05ky7yl31lmsgpiyyi554ch8aqvcm-aeson-1.5.6.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7xblky3nas5jz3qrzy0s1zl6clqbmq8y-attoparsec-0.13.2.5.drv' failed to build
error: 1 dependencies of derivation '/nix/store/r9hvzqps28dx0qzq3x4bxpf425pbgfvq-conduit-1.3.4.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/phvp3061k1gg17cb4ibix5c0dhjq589p-quickcheck-instances-0.3.25.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d73v8xk3s0g9xdrkvsiwri4rj0fr3z88-tree-diff-0.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6mg6k8x0y103dvw2xsnajc9kc21zac5m-pandoc-2.12.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vwpfi89m4kkhbaknvbhb158zrq2x2j1l-exa-unstable-2021-01-14.drv' failed to build
```

**Notify maintainers**
@peti 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.24-hardened1, NixOS, 21.05pre278406.d3f7e969b98 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210317_8a5203d`
 - channels(root): `"nixos-21.05pre278406.d3f7e969b98"`
 - channels(nequi): `"home-manager, nixpkgs-21.05pre278778.c0e5d0b73af"`
 - nixpkgs: `/home/nequi/.nix-defexpr/channels/nixpkgs`

**Comment by sternenseemann** [comment]
`vector` builds on `haskell-updates` again, so this should be fixed whenever #116965 is merged. We are currently waiting on hydra finishing the channel bumps for the OpenSSL security update.

**Comment by dotlambda** [comment]
fixed in master
